src/PolygonClipper/PolygonClipper.cs

```diff
@@ -569,8 +569,12 @@ private static int PossibleIntersection(
         }
 
         // The line segments associated with le1 and le2 overlap.
-        // TODO: Rewrite this to avoid allocation.
-        List<SweepEvent> events = new(4);
+        SweepEvent? first = null;
```

Comment:
I would attack this a different way by creating a span outside of the loop and passing it to the private method as a parameter.